l2tdevtools/dependency_writers/gift_ppa.py

```diff
@@ -72,35 +68,47 @@ class GIFTPPAInstallScriptWriter(interface.DependencyFileWriter):
       'fi',
       '']  # yapf: disable
 
-  def Write(self):
-    """Writes a gift_ppa_install.sh file."""
+  def _Write(self, python_dependencies, python_version_string='PYTHON2'):
+    """Writes a gift_ppa_install.sh file.
+
+    Args:
+      python_dependencies (list[str]): list of names of python dependencies.
+      python_version_string (Optional[str]): Python version.
+    """
     file_content = []
     file_content.extend(self._FILE_HEADER)
 
-    python2_dependencies = self._dependency_helper.GetDPKGDepends(
-        exclude_version=True, python_version=2)
-
     libyal_dependencies = []
-    for index, dependency in enumerate(python2_dependencies):
+    for index, dependency in enumerate(python_dependencies):
       if index == 0:
-        file_content.append('PYTHON2_DEPENDENCIES="{0:s}'.format(dependency))
-      elif index + 1 == len(python2_dependencies):
+        file_content.append('{0:s}_DEPENDENCIES="{1:s}'.format(
+            python_version_string, dependency))
+      elif index + 1 == len(python_dependencies):
         file_content.append('                      {0:s}";'.format(dependency))
       else:
         file_content.append('                      {0:s}'.format(dependency))
 
-      if dependency.startswith('lib') and dependency.endswith('-python'):
-        dependency, _, _ = dependency.partition('-')
-        libyal_dependencies.append(dependency)
+      if python_version_string == 'PYTHON2':
```

Comment:
Why not determine the suffix first:
```
if python_version_string == 'PYTHON2':
  dependency_suffix = '-python'
else:
  dependency_suffix = '-python3'
```

And then do the check using the suffix:
```
    if dependency.startswith('lib') and dependency.endswith(dependency_suffix):
      dependency, _, _ = dependency.partition('-')
      libyal_dependencies.append(dependency)
```

Reply:
Sure, if you prefer. Changed this.